Docs: Refactor 'Get started' section (#484)
grafana/plugin-tools: docusaurus/docs/get-started/get-started.mdx

import InstallPNPM from '@snippets/createplugin-install.pnpm.md';
import InstallYarn from '@snippets/createplugin-install.yarn.md';

# Get started

Welcome to the world of Grafana plugin creation, where you can enhance Grafana's foundational features. In this guide, you'll learn how to get started by scaffolding a plugin, running it in an efficient development environment, and using its basic features.

## Quick Start

Scaffold a new plugin with a single command! Run the following and answer the prompts:

> <CodeSnippets

Reviewer: <img width="337" alt="Screenshot 2023-10-26 at 19 18 42" src="https://github.com/grafana/plugin-tools/assets/19860021/77bc0393-43c9-4dc6-a056-2464f4dc44cc">
<img width="400" alt="Screenshot 2023-10-26 at 19 18 40" src="https://github.com/grafana/plugin-tools/assets/19860021/cdac0e34-4320-4ef0-9d05-0070345f88fc">

Not sure if we can get npx highlighted in the same manner? @jackw

Author: They're coded the same way, so that is odd. @jackw ?

  snippets={[
    { component: ScaffoldNPM, label: 'npm' },
    { component: ScaffoldPNPM, label: 'pnpm' },
    { component: ScaffoldYarn, label: 'yarn' },
  ]}
  groupId="package-manager"
  queryString="current-package-manager"
/>

## Why create a Grafana plugin?

Grafana plugin development allows you to create many different types of user experiences using [supported](/introduction/grafana-plugin-sdk-for-go) features. For example, you can make:

- **Panel plugins** - new ways of visualizing data
- **Data-source plugins** - connections to a new database or other source of data
- **App plugins** - integrated out-of-the-box experiences

:::tip

If this is your first time creating a plugin, we recommand that you familiarize yourself with the fundamentals of plugin types, backend plugins, data frames, and other key concepts. For more, refer to the [Introduction to Grafana plugin development](/introduction/).

:::

## Use plugin tools to develop your plugins faster

> Grafana's plugin tools offer an officially supported way to extend Grafana's core functionality. We have designed these tools to help you to develop your plugins faster with a modern build setup and zero configuration.

Reviewer: ```suggestion
Grafana's plugin tools offer an officially supported way to extend Grafana's core functionality. We have designed these tools to help you to develop your plugins faster with a modern build setup and zero additional configuration required.
```

Reviewer: just a thought

Author: yup.

└── tsconfig.json
```

For more information about these files, refer to [Folder structure](get-started/folder-structure/).

### Step 3: Run your plugin in Docker

With the `create-plugin` tool, you can use a Docker container to simplify the configuration, loading, and development processes. For more information, refer to [Set up development environment](get-started/set-up-development-environment/).

### Step 4: Use CLI commands

> After you have installed `create-plugin`, you can run built-in commands in the shell. Refer to the section below.

Reviewer 2: nit: I would add a link like:

```suggestion
After you have installed `create-plugin`, you can run built-in commands in the shell. Refer to the [section below](#frequently-used-cli-commands).
```

## Frequently used CLI commands

> Use the CLI for essential tasks of plugin development.

Reviewer 2: I would add a note here to tell that these are `npm` or `yarn` commands (so the full comand is for example `npm run build`)

Author: done.

### `run build`

Compiles and bundles the project using Webpack in production mode.

> ### `run dev`

Reviewer 2: what about `mage -v`? It's more important than the rest if the plugin has a backend part.

Author: Good catch. added mage.

└── tsconfig.json
```

For more information about these files, refer to [Folder structure](get-started/folder-structure/).

### Step 3: Run your plugin in Docker

With the `create-plugin` tool, you can use a Docker container to simplify the configuration, loading, and development processes. For more information, refer to [Set up development environment](get-started/set-up-development-environment/).

> ### Step 4: Use CLI commands

Reviewer: There's a fair bit of overlap between steps 3 and 4. I'd be inclined to remove step 4 and just detail the key commands in step 3:
- install (npm i)
- build the frontend in development mode (npm run dev)
- build the backend (optional) (mage -v build:linux)
- run the Grafana server (npm run server)

Author: done.

After you have installed `create-plugin`, you can run built-in commands in the shell. Refer to the [section below](#frequently-used-cli-commands).

## Frequently used CLI commands

Use the CLI for essential tasks of plugin development, employing `npm`, `pnpm`, or `yarn` (for example, `npm run build`).

### `mage -v`

Builds backend plugin binaries for Linux, Windows and Darwin.

### `run build`

Compiles and bundles the project using Webpack in production mode.

### `run dev`

Runs Webpack in watch mode for development, continually monitoring for changes.

### `run typecheck`

Performs a type-checking process on the frontend code using TypeScript.

### `run server`

> Launches the https://grafana.com/developers/plugin-tools/get-started/set-up-development-environment using Docker.

Reviewer: ```suggestion
Launches the [Grafana development server](https://grafana.com/developers/plugin-tools/get-started/set-up-development-environment) using Docker.
```

Author: done.

Author: done.

### `run sign`

Signs the Grafana plugin using the latest version of `@grafana/sign-plugin`.

:::info

For a complete list of available CLI commands, refer to the [CLI reference](/get-started/prompt-cli-reference#cli-commands).

:::

## Next steps

- Start your plugin journey with one of our [plugin development tutorials](/tutorials/).
- Learn how to [develop and extend](/create-a-plugin/) its functionality.

> - Review the [Examples](https://github.com/grafana/grafana-plugin-examples), [sign](/publish-a-plugin/sign-a-plugin) to learn about good practices.

Reviewer: ```suggestion
- Review the [Examples](https://github.com/grafana/grafana-plugin-examples), to learn about good practices.
- Learn how to package(link), sign(link) and publish(link) your plugin to the Grafana plugin catalog(link). 
```
In the last bullet, sign feels out of place, also not sure why Examples is capitalised?

Author: good suggestions. done. Examples shouldn't have been capitalized; we capitalize the titles of help topics but in this case it wasn't a help topic.
